syllabus/task_space/task_space.py

```diff
@@ -39,6 +46,18 @@ def _make_task_encoder(self, space, tasks):
             decoders = [r[1] for r in results]
             encoder = lambda task: [e(t) for e, t in zip(encoders, task)]
             decoder = lambda task: [d(t) for d, t in zip(decoders, task)]
+
+        elif isinstance(space, MultiDiscrete):
+            if isinstance(space, MultiDiscrete):
+                assert space.nvec.ndim == len(tasks), f"Number of steps in a tasks ({space.nvec.ndim}) must match number of discrete options ({len(tasks)})"
+            
+            combinations = [p for p in itertools.product(*list)]
+            self._encode_map = {task: i for i, task in enumerate(combinations)}
+            self._decode_map = {i: task for i, task in enumerate(combinations)}
+            temp = ",".join(str(element) for element in task)
+            encoder = lambda task: self._encode_map[temp] if temp in self._encode_map else Nonefmp 
+            decoder = lambda task: self._decode_map[temp] if task in self._decode_map else None
```

Comment:
This is an interesting approach. I'm not sure if it's better to convert the MultiDiscrete to a tuple or a unique int like you've done here. This might actually be better for compatibility with curricula that only support discrete task lists (most of them). That being said, the decoded version should use task names, not just arbitrary integers. We should be able to encode ("Small", "Easy", 11) to an integer then decode it back to that original tuple.